### ISSUE TITLE
JIT: Pred block link in handler PHI args are of questionable utility

### ISSUE THREAD

**Issue by jakobbotsch** [dotnet/runtime#94981]
Ever since #85546 we are keeping block links as part of phi args. However, for handlers this information seems questionable for a few reasons:
1. RBO cannot actually do any phi disambiguation for handlers anyway, so they are unused
2. Handlers can have tons of predecessors, so this can lead to quite large phis
3. Even if we wanted to use it, SSA does not build them correctly today, so that would need to be fixed first

By 3 I mean the following example:
```csharp
static void Main(string[] args)
{
    int x = 10;
    try
    {
        if (Environment.TickCount < 0)
        {
            Console.WriteLine(0);
        }

        // BB04
        M();
        x = 20;
        M();
    }
    catch
    {
        Console.WriteLine(x);
    }
}

[MethodImpl(MethodImplOptions.NoInlining)]
private static void M()
{
}
```

Here the handler's phi looks like
```scala
N005 (  0,  0) [000020] DA---------                         ▌  STORE_LCL_VAR int    V01 loc0         d:3
N004 (  0,  0) [000019] -----------                         └──▌  PHI       int   
N001 (  0,  0) [000023] ----------- pred BB03                  ├──▌  PHI_ARG   int    V01 loc0         u:2
N002 (  0,  0) [000022] ----------- pred BB04                  ├──▌  PHI_ARG   int    V01 loc0         u:4
N003 (  0,  0) [000021] ----------- pred BB02                  └──▌  PHI_ARG   int    V01 loc0         u:2
```
which does not leave the possibility that `x` could have more than one value when coming from `BB04`. The essential "problem" is that this code probably ought to run for all blocks in try regions:
https://github.com/dotnet/runtime/blob/83f116627e2548c7a55b3701bd5448570b84a784/src/coreclr/jit/ssabuilder.cpp#L1230-L1233

But, overall it seems unlikely we'll ever be able to make use of the information, so I propose that we allow the pred block links to be null for handler blocks (i.e. go back to before #85546 for that case). Thoughts @AndyAyersMS @dotnet/jit-contrib?

**Comment by BruceForstall** [comment]
Makes sense to me: if the information is incorrect and unused (and expensive), and handler entry needs special handling anyway, seems like no need to keep it.

**Comment by AndyAyersMS** [comment]
> But, overall it seems unlikely we'll ever be able to make use of the information, so I propose that we allow the pred block links to be null for handler blocks (i.e. go back to before #85546 for that case). Thoughts @AndyAyersMS @dotnet/jit-contrib?

Seems reasonable, I think we can special case handler entry phis.

**Comment by amanasifkhalid** [comment]
This doesn't seem critical to .NET 10 -- pushing this out
